internal/store/memory/gateway.go

```diff
@@ -60,6 +60,7 @@ func (g *gatewayState) TryBind(ctx context.Context, route store.Route) {
 			if err != nil {
 				// consider each route distinct for the purposes of binding
 				g.logger.Debug("error binding route to gateway", "error", err, "route", route.ID())
+				l.RemoveRoute(route.ID())
```

Comment:
Will we always get an `err` value when `canBind == false`? If not,  should we also  call `RemoveRoute` below when `!canBind`?

Reply:
> As an aside, the (bool, error) return of CanBind seems redundant and makes it prone to misuse - should it be changed to just one of those values instead of the tuple?

I would **much** prefer a single return value if `canBind` is always `true` unless `err != nil`.

Reply:
Started trying to rip this out and ran into the following bit which does seem to return `false, nil` in some cases https://github.com/hashicorp/consul-api-gateway/blob/2362a739269ac7778ed41a8772602d4914dfe553/internal/k8s/reconciler/listener.go#L339-L388